src/main/scala/piconot/internal/Empty.scala

```diff
@@ -0,0 +1,46 @@
+package piconot.internal
+
+import picolib.semantics._
+
+import scala.language.postfixOps
+
+/**
+  * Created by tmf on 10/9/16.
+  */
+object EmptyBot extends PicoWhile("resources" / "empty.txt") {
+
+  WhileOuter (free(West)) {
```

Comment:
For your internal DSL, I'm wondering why you have, for ex., WhileOuter (free(West)). I think that you could simplify this to WhileOuter free(West), maybe?